crates/ruff_python_parser/src/lexer.rs

```diff
@@ -241,7 +242,7 @@ impl<'source> Lexer<'source> {
             "yield" => Tok::Yield,
             _ => {
                 return Ok(Tok::Name {
-                    name: text.to_string(),
+                    name: SmolStr::new(text),
```

Comment:
How much work would it be, in your opinion, to separate the `SmolStr` changes out to a separate PR, with this PR as the upstream? It feels like they're slightly independent from the changes to the parser and error recovery etc., and so separating them out would make it easier to merge this PR and the `SmolStr` PR (since they can be reviewed and considered independently), and easier to understand the impact of the `SmolStr` change.

Reply:
Probably won't be much work, most of the changes related to `SmolStr` are in one commit.

I just need to learn how to do that, any help would appreciated.

Reply:
You can `git rebase -i main` then just "drop" that commit, noting its hash. Then you can create a new branch from your parser branch, and `git cherry-pick <commit>` the commit onto it.

Reply:
@zanieb - the problem with this suggestion (not sure if you have ideas…) is that I don’t believe @LaBatata101 will be able to create a PR on Ruff that uses this current PR as a branch, since this branch isn’t in Ruff, it’s in a fork.

Reply:
Ah true. You can create a second pull request with `main` as the base and it'll just duplicate all of the changes from here until this is merge,  you can just wait until this is merged to open the second pull request, or you can open it over on your fork and link to it if you want feedback sooner 🤷‍♀️